pylearn2/sandbox/rnn/space/__init__.py

```diff
@@ -181,11 +175,16 @@ def get_origin_batch(self, batch_size, dtype=None):
 
     @wraps(space.Space.make_theano_batch)
     def make_theano_batch(self, name=None, dtype=None, batch_size=None):
-
         dtype = self._clean_dtype_arg(dtype)
-        broadcastable = [False] * 3
-        broadcastable[1] = (batch_size == 1)
-        broadcastable[2] = (self.dim == 1)
+        if not isinstance(self.space, Conv2DSpace):
```

Comment:
Here, it would be better to whitelist `VectorSpace` and maybe `IndexSpace`, rather than assuming that everything except `Conv2DSpace` has 2 dimensions with the first one being the batch.